### PR TITLE
Completed Javascript-Typing-Tutor

### PR DESCRIPTION
**Pull request by Henry-Nichols**
Screen Shot 2021-04-06 at 9 41 26 AM

### CODE REVIEW

File: javascript-typing-tutor/main.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+var restart = document.querySelector('button');
+var typer = document.getElementById('phrase');
+var arr = typer.textContent.split('');
```
Comment (codingcodymiller) [inline javascript-typing-tutor/main.js:3]: You can store the phrase in a variable here in javascript, and just generate the spans and append when the page loads, so you don't need to get your data from an HTML element before you can do the work you want to do.

File: javascript-typing-tutor/main.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    spanArr[position].classList.add('correct');
+
+    spanArr[position].classList.remove('underline');
+    spanArr[position+1].classList.add('underline');
+
+    typer.innerHTML = '';
+
+    for(var i = 0; i < arr.length; i++){
+      typer.appendChild(spanArr[i]);
+    }
```
Comment (codingcodymiller) [inline javascript-typing-tutor/main.js:26]: You shouldn't need to destroy all elements and rebuild them every time a letter is typed. Instead, when you generate your array of spans when the page loads, you should append those to the DOM right away and just update the class list on the current letter whenever the user types.

File: javascript-typing-tutor/main.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  }else{
+    spanArr[position].classList.add('incorrect');
+
+  }
+
+});
+
+restart.addEventListener('click', function(e) {
+  window.location.reload();
+})
```
Comment (codingcodymiller) [inline javascript-typing-tutor/main.js:39]: You should **_never_** use `location.reload`. It entirely refreshes the page, so the whole page needs to get built again from scratch, which is highly inefficient. Rather, if somebody wants to play again, you just remove all the classes from the letters that are already applied, and reset the value of your variable for what `position` the user is at, so they can just play again without needing to refresh.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Great start here, man!! A couple notes for you:

File: javascript-typing-tutor/main.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+restart.addEventListener('click', function (e) {
+  var iterator = spanArr.values();
+  for (var el of iterator) {
+    el.classList.remove('correct');
+    el.classList.remove('incorrect');
+    el.classList.remove('underline');
+  }
+});
+=======
+restart.addEventListener('click', function(e) {
```
Comment (thebearingedge) [inline javascript-typing-tutor/main.js:80]: Please clean up the merge conflict in your JS and then add, commit, and push again.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by thebearingedge — APPROVED** [review]
Stylish!

**Comment by Henry-Nichols** [comment]
> Stylish!

Thank you Tim!

**Pull request merged**